src/assets/Azure.Core.Shared/OperationInternalOfT.cs

```diff
@@ -333,6 +376,17 @@ internal interface IOperation<T>
         /// </list>
         /// </returns>
         ValueTask<OperationState<T>> UpdateStateAsync(bool async, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// To get the Id of the operation for rehydration purpose.
+        /// </summary>
+        string GetOperationId();
+    }
+
+    internal interface IOperationSource<T>
+    {
+        T CreateResult(Response response, CancellationToken cancellationToken);
+        ValueTask<T> CreateResultAsync(Response response, CancellationToken cancellationToken);
```

Comment:
Moved `IOperationSource` from `Generator.Shared` to `Azure.Core.Shared`. There are 2 SDKs which pinged generator version complained about this change.